src/actors/runtime/exitcode/vm_exitcodes.go

```diff
@@ -65,27 +65,41 @@ const (
 	InsufficientPledgeCollateral
 )
 
+type ExitCode struct {
+	Success          struct{}
+	SystemError      SystemErrorCode
+	UserDefinedError UserDefinedErrorCode
+}
```

Comment:
Thanks for this faithful port, but it's too complex. Can you replace all of this with a single `type ErrorCode int64` and some helper methods? I think we can remove the concept of UserDefinedErrorCode.